server/server.js

```diff
@@ -172,6 +190,106 @@ app.post('/genotet/upload', upload.single('file'), function(req, res) {
   });
 });
 
+app.use(cookieParser());
+
+app.post('/genotet/user', function(req, res) {
+  console.log('POST user');
+
+  var type = req.body.type;
+  var userInfo = {};
+  var response;
+  switch (type) {
+    case user.QueryType.SIGHUP:
+      data = {
+        email: req.body.email,
+        username: req.body.username,
+        password: req.body.password,
+        confirmed: req.body.confirmed
+      };
+      response = user.signUp(userPath, data);
+      break;
+    case user.QueryType.SIGNIN:
+      data = {
+        username: req.body.username,
+        password: req.body.password
+      };
+      response = user.signIn(userPath, data);
+      break;
+
+    // Undefined type, error
+    default:
+      console.error('invalid query type');
+      data = {
+        error: {
+          type: 'query',
+          message: 'invalid query type'
+        }
+      };
+  }
+  res.header('Access-Control-Allow-Origin', '*');
+  if (data.error) {
+    console.log(data.error);
+    res.status(500).json(data.error);
+  } else if (response.success) {
+    // Get session ID from database. Regenerate if it is expired.
+    var username = req.body.username;
+    var findUserInfo = function(db, callback) {
+      var cursor = db.collection('session').find({'username': username});
+      var result = [];
+      cursor.each(function(err, doc) {
+        assert.equal(err, null);
+        if (doc != null) {
+          result.push(doc);
+        } else {
+          callback();
+        }
+      });
+      callback = function() {
+        var hasValidSession = false;
+        var sessionIndex = -1;
+        if (result.length != 0) {
+          for (var i = 0; i < result.length; i++) {
+            if (new Date().getTime() < result[i].expireDate) {
+              sessionIndex = i;
+              hasValidSession = true;
+              break;
+            }
+          }
+        }
+        if (result.length == 0 || !hasValidSession) {
+          // Don't have username or have username but don't have valid session
+          var cookie = {
+            username: username,
+            sessionID: utils.guid(),
+            expireDate: new Date().getTime() + 24 * 60 * 1000
```

Comment:
What you stored is not a date but a timestamp. Probably a better attribute name is 'expire' or 'expiration'.